python/kserve/kserve/protocol/infer_type.py

```diff
@@ -0,0 +1,544 @@
+            HTTP body after the JSON object.
+        Raises
+        ------
+        InferenceServerException
+            If failed to set data for the tensor.
+        """
+        if not isinstance(input_tensor, (np.ndarray,)):
+            raise_error("input_tensor must be a numpy array")
+
+        dtype = from_np_dtype(input_tensor.dtype)
+        if self._datatype != dtype:
+            raise_error(
+                "got unexpected datatype {} from numpy array, expected {}".format(dtype, self._datatype))
+        valid_shape = True
+        if len(self._shape) != len(input_tensor.shape):
+            valid_shape = False
+        else:
+            for i in range(len(self._shape)):
+                if self._shape[i] != input_tensor.shape[i]:
+                    valid_shape = False
+        if not valid_shape:
+            raise_error(
+                "got unexpected numpy array shape [{}], expected [{}]".format(
+                    str(input_tensor.shape)[1:-1],
+                    str(self._shape)[1:-1]))
+
+        if not binary_data:
+            self._parameters.pop('binary_data_size', None)
+            self._raw_data = None
+            if self._datatype == "BYTES":
+                self._data = []
+                try:
+                    if input_tensor.size > 0:
+                        for obj in np.nditer(input_tensor,
+                                             flags=["refs_ok"],
+                                             order='C'):
+                            # We need to convert the object to string using utf-8,
+                            # if we want to use the binary_data=False. JSON requires
+                            # the input to be a UTF-8 string.
+                            if input_tensor.dtype == np.object_:
+                                if type(obj.item()) == bytes:
+                                    self._data.append(
+                                        str(obj.item(), encoding='utf-8'))
+                                else:
+                                    self._data.append(str(obj.item()))
+                            else:
+                                self._data.append(
+                                    str(obj.item(), encoding='utf-8'))
+                except UnicodeDecodeError:
+                    raise_error(
+                        f'Failed to encode "{obj.item()}" using UTF-8. Please use binary_data=True, if'
+                        ' you want to pass a byte array.')
+            else:
+                self._data = [val.item() for val in input_tensor.flatten()]
+        else:
+            self._data = None
+            if self._datatype == "BYTES":
+                serialized_output = serialize_byte_tensor(input_tensor)
+                if serialized_output.size > 0:
+                    self._raw_data = serialized_output.item()
+                else:
+                    self._raw_data = b''
```

Comment:
```suggestion
```

Reply:
eh actually just reread this and idk, you could remove this else like I suggested but its fine either way